echoregions/regions2d/regions2d.py

```diff
@@ -84,50 +77,28 @@ def min_depth(self, val: Union[int, float]) -> None:
                 raise ValueError("min_depth cannot be greater than max_depth")
         self._min_depth = float(val) if val is not None else val
 
-    @property
-    def offset(self) -> Union[int, float]:
-        """Get the depth offset to apply to y values"""
-        return self._offset
-
-    @offset.setter
-    def offset(self, val: Union[int, float]) -> None:
-        """Set the depth offset to apply to y values"""
-        self._offset = float(val)
-
-    @property
-    def plotter(self) -> Regions2DPlotter:
-        if self._plotter is None:
-            if not self.data:
-                raise ValueError(
-                    "Input file has not been parsed; call `parse_file` to parse."
-                )
-            self._plotter = Regions2DPlotter(self)
-        return self._plotter
-
-    def parse_file(self, offset: Union[int, float] = 0) -> None:
-        # TODO make use of offset.
-        """Parse the EVR file as a DataFrame into `Regions2D.data`"""
-        self.data = self._parser.parse_file()
-        self.replace_nan_depth()
-        self.adjust_offset()
-
-    def to_csv(self, save_path: str = None, **kwargs) -> None:
-        """Convert an EVR file to a CSV file
+    def to_csv(self, save_path: bool = None) -> None:
+        """Save a Dataframe to a .csv file
 
         Parameters
         ----------
         save_path : str
-            Path to save csv file to
-        convert_time : bool, default False
-          Convert times in the EV datetime format to numpy datetime64.
-        kwargs : keyword arguments
-            Additional arguments passed to `Regions2D.parse_file`
+            path to save the CSV file to
         """
-        if self.data is None:
-            self.parse_file(**kwargs)
-        self._parser.to_csv(self.data, save_path=save_path, **kwargs)
+        if not isinstance(self.data, DataFrame):
+            raise TypeError(
+                f"Invalid ds Type: {type(self.data)}. Must be of type DataFrame."
+            )
```

Comment:
Similar to the Lines case, I don't think we need to check the type of `self.data` here since it is set internally.